### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by amexabee**
- Create model classes for all entities 
- Set up associations between models
- Create custom methods
  - A method that returns the 3 most recent posts for a given user.
  - A method that updates the posts counter for a user.
  - A method that returns the 5 most recent comments for a given post.
  - A method that updates the comments counter for a post.
  - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
```
Comment (Ismailco) [inline app/models/user.rb:1]: - [ ] Kindly make sure to set up associations between models, the `user` has many `posts`, `comments`, and `likes`. also don't forget to add a foreign key to `author_id`

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require_relative 'comment'
```
Comment (Ismailco) [inline app/models/post.rb]: - [ ] Same here the `post` has many `comments` and `likes`, and it belongs_to `author`, kindly make sure to set up these associations so your modules work properly

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
```
Comment (Ismailco) [inline app/models/like.rb:1]: - [ ] The `likes` belong to the `author` and post, kindly make sure to set up associations for this so it works properly

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
```
Comment (Ismailco) [inline app/models/comment.rb:1]: - [ ] The `comment` belong to the `author` and post, kindly make sure to set up associations for this so it works properly

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @amexabee, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ Work documented professionally
✔️  No linter errors
✔️ Used Gitflow

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by franclobo — APPROVED** [review]
Hi @amexabee, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors ✔️ 
- Methods are created correctly ✔️ 
- Modals are completed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**